File: Leetcode 2849. Determine if a Cell Is Reachable at a Given Time/lc 2849.cpp
class Solution {
public:
    bool isReachableAtTime(int sx, int sy, int fx, int fy, int t) {
        int rowDiff=abs(sx-fx);
        int colDiff=abs(sy-fy);
        int minOps=max(rowDiff,colDiff);
        if(t<minOps){
            return false;
        }else if(minOps==0 && t==1){
            return false;
        }else{
            return true;
        }
    }
};

File: Leetcode 2849. Determine if a Cell Is Reachable at a Given Time/lc 2849.java
class Solution {
    public boolean isReachableAtTime(int sx, int sy, int fx, int fy, int t) {
        int colDiff=Math.abs(sx-fx);
        int rowDiff=Math.abs(sy-fy);
        int maxCount=Math.max(rowDiff,colDiff);
        if(maxCount==0 && t==1)return false;
        if(maxCount<=t){
            return true;
        }
        return false;
        
    }
}